packages/govuk-frontend/src/govuk/components/header/_index.scss

```diff
@@ -166,8 +174,16 @@
   .govuk-header__menu-button {
     @include govuk-font($size: 16);
     position: absolute;
-    top: govuk-spacing(4);
+    // calculate top offset by:
+    // - getting the vertical spacing for the top and the bottom of the header
+    // - adding that to the crown height
+    // - dividing it by 2 so you have the vertical centre of the header
+    // - subtracting half the height of the menu button
+    top: (((govuk-spacing($govuk-header-vertical-spacing-value) * 2) + $govuk-header-crown-height) / 2) -
+      ($govuk-header-menu-button-height / 2);
     right: 0;
+    max-width: $govuk-header-menu-button-width;
```

Comment:
Looks good @owenatgov

My only hesitation would be around the new `max-width: 80px` restriction.

Teams can customise `params.menuButtonLabel` so full width with a logo-sized safe area would be better? Probably a lot easier to do once https://github.com/alphagov/govuk-frontend/issues/1739 lands

<img width="460" alt="Menu button with long name" src="https://github.com/alphagov/govuk-frontend/assets/415517/f18d7d47-c6f2-44f3-a76e-989047967e7a">

Reply:
I made a comment in https://github.com/alphagov/govuk-frontend/pull/4150/commits/e9d62b4d19f5a7554eb948e067d75a66f3db6798 that the only instance of a service not using the text "menu" that's publicly available is the Welsh for menu: "Dewislen". However, that's a good point, we could just turn the calculation the other way around. I had a go at "shrink wrapping" the logo's width but was having trouble with it. I also need to have a think about how we could set the padding to the rest of the width container; I'm sure I've managed to do this before? Let me muck around with this. If you have any ideas I am very open to them.

Reply:
Here's what I was thinking

Including margins, the logo occupies 183px width (or 191px using Arial)

Will need to adjust for potential text-only zoom, but given a 200px "safe area" we could:

```patch
--- A/packages/govuk-frontend/src/govuk/components/header/_index.scss
+++ B/packages/govuk-frontend/src/govuk/components/header/_index.scss
@@ -10,8 +10,8 @@
   // This crown height is only used to calculate top offset of mobile menu button
   // as the crown svg height is the only thing that controls the height of the header
   $govuk-header-crown-height: 30px;
+  $govuk-header-logo-width: 200px;
   $govuk-header-menu-button-height: 20px;
-  $govuk-header-menu-button-width: 80px;
 
   .govuk-header {
     @include govuk-font($size: 16);
@@ -83,6 +83,7 @@
     @include govuk-typography-common;
     @include govuk-link-style-inverse;
 
+    max-width: $govuk-header-logo-width;
     text-decoration: none;
 
     &:hover {
@@ -151,10 +152,6 @@
 
   .govuk-header__logo {
     @include govuk-responsive-margin($govuk-header-vertical-spacing-value, "bottom");
-    // Protect the absolute positioned menu button from overlapping with the
-    // logo with right padding using the button's width
-    padding-right: $govuk-header-menu-button-width;
-
     @include govuk-media-query($from: desktop) {
       width: 33.33%;
       padding-right: $govuk-gutter-half;
@@ -182,13 +179,16 @@
     top: (((govuk-spacing($govuk-header-vertical-spacing-value) * 2) + $govuk-header-crown-height) / 2) -
       ($govuk-header-menu-button-height / 2);
     right: 0;
+    width: 100%;
-    max-width: $govuk-header-menu-button-width;
     min-height: $govuk-header-menu-button-height;
     margin: 0;
+    // Protect the absolute positioned menu button from overlapping with the
+    // logo with left padding using the logo's width
+    padding: 0 0 0 $govuk-header-logo-width;
-    padding: 0;
     border: 0;
     color: govuk-colour("white");
     background: none;
+    text-align: right;
     cursor: pointer;
 
     &:hover {
```

Reply:
Downsides are that the button's clickable area will grow to fill non-logo space

Not a bad thing at mobile size, but could fix the width at tablet size perhaps

Reply:
Pushed up a branch [header-menu-button-position-refactor-colin](https://github.com/alphagov/govuk-frontend/compare/header-menu-button-position-refactor-colin) with commit https://github.com/alphagov/govuk-frontend/commit/4b0273849b36a881bf590f55491b0b0c43b30316